File: app/domain/strategies/primer_clase_descuento_strategy.rb
require_relative '../interfaces/i_precio_strategy'

module Strategies
  class PrimerClaseDescuentoStrategy
    include Interfaces::IPrecioStrategy

    def calcular(precio_base, cliente)
      # 10% de descuento si es la primera reserva del cliente
      cliente.historial.empty? ? precio_base * 0.90 : precio_base
    end
  end
end

File: EduLink.Domain/Strategies/PrimerClaseDescuentoStrategy.cs
using EduLink.Domain.Entities;
using EduLink.Domain.Interfaces;

namespace EduLink.Domain.Strategies;

public class PrimerClaseDescuentoStrategy : IPrecioStrategy
{
    public decimal Calcular(decimal precioBase, Cliente cliente)
    {
        return cliente.Historial.Count == 0
            ? precioBase * 0.90m  // 10% de descuento
            : precioBase;
    }
}